pkg/destroy/aws/aws.go

```diff
@@ -184,6 +184,51 @@ func (o *ClusterUninstaller) Run() error {
 		}
 	}
 
+	if err := o.deleteInstanceProfileHack(awsSession); err != nil {
+		o.Logger.Debug(err)
+	}
+
+	return nil
+}
+
+// THIS IS A HORRIBLE HACK! Users keep ending up with all of their instance and the roles
+// deleted, but not the instance profiles. This horrific hack guesses the name of the
+// instance profiles and delete them. It's terrible. But it works even if something else
+// deleted artifacts behind our back.
+func (o *ClusterUninstaller) deleteInstanceProfileHack(awsSession *session.Session) error {
+	o.Logger.Debug(fmt.Sprintf("%v", o))
+	clusterID := ""
+	openshiftClusterID := ""
+	for _, filter := range o.Filters {
+		for key, value := range filter {
+			if strings.HasPrefix(key, "kubernetes.io/cluster/") {
+				clusterID = strings.TrimPrefix(key, "kubernetes.io/cluster/")
+				continue
+			}
+			if key == "openshiftClusterID" {
+				openshiftClusterID = value
+				continue
+			}
+		}
+		if openshiftClusterID != "" && clusterID != "" {
+			break
+		}
+	}
+	if clusterID == "" {
+		return fmt.Errorf("unable to determine clusterID in deleteInstanceProfileHack()")
+	}
+	if openshiftClusterID == "" {
+		return fmt.Errorf("unable to determine openshiftClusterID in deleteInstanceProfileHack()")
+	}
+	masterARN := fmt.Sprintf("arn:aws:iam::123456789012:instance-profile/%s-master-profile-%s", clusterID, openshiftClusterID)
```

Comment:
My preference would be to have a `deleteIAMInstanceProfileByName` function that is passed the name of the profile rather than making up an ARN with a fake account ID that the caller is trusting won't actually be used.